app/error.go

```diff
@@ -0,0 +1,180 @@
+package app
+
+// ErrorType defines the type of an error
+type ErrorType string
+
+const (
```

Comment:
Because of the rewrite you did, apperror.Validation is now app.Valudation which doesn't sound like an error

block of constants these are usually prefixed with a common name

In your case, I would use ErrCode or ErrorCode, or simply ErrorType

So Validation becomes ErrorCodeValidation or ErrorTypeValidation and so on for other constant from this block